### PR TITLE
Feature/pt 4298 transfer ownership

### PR DESCRIPTION
**Pull request by ifelsedeveloper**

### CODE REVIEW

File: upgrade-executor/src/actions/TimelockAdminshipTransferAndRenounceAction.sol [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+
+contract TimelockAdminshipTransferAndRenounceAction {
+    bytes32 public constant TIMELOCK_ADMIN_ROLE = keccak256("TIMELOCK_ADMIN_ROLE");
+    
+    function perform(address target, address newOwner) public payable {
```
Comment (evgenynacu) [inline upgrade-executor/src/actions/TimelockAdminshipTransferAndRenounceAction.sol:9]: Should it be payable?

File: upgrade-executor/tasks/transfer-ownership.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { AccessControlUpgradeable__factory } from "../typechain-types";
+
+export async function transferOwnership(
+  settings: IContractsTransfer,
+  newOwner: string,
+  signer: ethers.Signer
+) {
+  try {
+    // Transfer ownership of each contract in contracts
+    for (const [key, address] of Object.entries(settings.contracts)) {
```
Comment (evgenynacu) [inline upgrade-executor/tasks/transfer-ownership.ts:14]: It seems logic for transfering ownership of "contracts" and "nonupgradeable" is the same. Do we need both?

**Review by evgenynacu — CHANGES_REQUESTED** [review]

File: upgrade-executor/tasks/transfer-ownership.ts (hunk 2 above)
Comment (ifelsedeveloper) [reply upgrade-executor/tasks/transfer-ownership.ts:14]: Yes, We can have sections like ownable and accesscontrol

File: upgrade-executor/src/actions/TimelockAdminshipTransferAndRenounceAction.sol (hunk 1 above)
Comment (disenotov) [reply upgrade-executor/src/actions/TimelockAdminshipTransferAndRenounceAction.sol]: fixed

File: upgrade-executor/tasks/transfer-ownership.ts (hunk 2 above)
Comment (disenotov) [reply upgrade-executor/tasks/transfer-ownership.ts]: fixed

File: upgrade-executor/tasks/transfer-ownership.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { AccessControlUpgradeable__factory } from "../typechain-types";
+
+export async function transferOwnership(
+  settings: IContractsTransfer,
+  newOwner: string,
+  signer: ethers.Signer
+) {
+  try {
+    // Transfer ownership of each contract in contracts
+    for (const [key, address] of Object.entries(settings.ownable)) {
```
Comment (evgenynacu) [inline upgrade-executor/tasks/transfer-ownership.ts:14]: formatting is different across the file

File: upgrade-executor/tasks/transfer-ownership.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      await tx.wait(5)
+      console.log(`Ownership of ${key} transferred successfully`);
+    }
+
+    // Transfer ownership of each access contract
+    for (const [key, address] of Object.entries(settings.accessControl)) {
+        console.log(`Add admin role of ${key} to ${newOwner}`);
+        const contract = AccessControlUpgradeable__factory.connect(address, signer);
+        const TIMELOCK_ADMIN_ROLE = ethers.utils.id("TIMELOCK_ADMIN_ROLE");
+        const tx = await contract.grantRole(TIMELOCK_ADMIN_ROLE, newOwner)
```
Comment (evgenynacu) [inline upgrade-executor/tasks/transfer-ownership.ts:28]: what happens if this tx fails and next is ok? we just lose control I think?

**Review by evgenynacu — CHANGES_REQUESTED** [review]

File: upgrade-executor/tasks/transfer-ownership.ts (hunk 4 above)
Comment (evgenynacu) [inline upgrade-executor/tasks/transfer-ownership.ts]: Do we use only TIMELOCK_ADMIN_ROLE? Why?

**Review by evgenynacu — CHANGES_REQUESTED** [review]

**Pull request merged**